### PR TITLE
Vet clinic database: query multiple tables

### PR DESCRIPTION
**Pull request by Dorcas126**
In this project, I  will add some new tables and add foreign key columns to the existing animals table to model one-to-many relationships. Afterward, I will use JOIN to query the data.

# Project Description

- Create a table named owners with the following columns:
    - id: integer (set it as autoincremented PRIMARY KEY)
    - full_name: string
    - age: integer
- Create a table named species with the following columns:
    - id: integer (set it as autoincremented PRIMARY KEY)
    - name: string
 - Modify animals table:
      - Make sure that id is set as autoincremented PRIMARY KEY
      - Remove column species
      - Add column species_id which is a foreign key referencing species table
      - Add column owner_id which is a foreign key referencing the owner's table
 - Insert the following data into the owner's table:
     - Sam Smith 34 years old.
     - Jennifer Orwell 19 years old.
     - Bob is 45 years old.
     - Melody Pond 77 years old.
     - Dean Winchester 14 years old.
     - Jodie Whittaker 38 years old.
 - Insert the following data into the species table:
    - Pokemon
    - Digimon
- Modify, and insert animals so they include the species_id value:
    - If the name ends in "mon" it will be Digimon
    - All other animals are Pokemon
- Modify, and insert animals to include owner information (owner_id):
  - Sam Smith owns Agumon.
  - Jennifer Orwell owns Gabumon and Pikachu.
  - Bob owns Devimon and Plantmon.
  - Melody Pond owns Charmander, Squirtle, and Blossom.
  - Dean Winchester owns Angemon and Boarmon.
- Write queries (using JOIN) to answer the following questions:
   - What animals belong to Melody Pond?
   - List of all animals that are pokemon (their type is Pokemon).
   - List all owners and their animals, remember to include those that don't own any animal.
   - How many animals are there per species?
   - List all Digimon owned by Jennifer Orwell.
   - List all animals owned by Dean Winchester that haven't tried to escape.
   - Who owns the most animals?

image

image

image

image

image

image

image

image

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -158,3 +158,65 @@ FROM animals
...
+
+UPDATE animals
+SET owner_id = 
+    (SELECT id FROM owners WHERE full_name = 'Bob')
+    WHERE name IN ('Devimon', 'Plantmon');
+
+UPDATE animals
+SET owner_id = 
+    (SELECT id FROM owners WHERE full_name = 'Dean Winchester')
+    WHERE name IN ('Angemon', 'Boarmon');
```
Comment (V-Blaze) [inline queries.sql]: - [ ] You have done great so far :+1: :+1: But there are a few changes you need to make before moving to the next milestone. :+1: :100: 
- [ ] These SQL codes are adding data to the database and therefore should be in the `data.sql` file and not in the queries.sql. I suggest you move these to the correct file as required. :smile: :wave:

File: queries.sql [hunk 2]
```diff
@@ -158,3 +158,65 @@ FROM animals
...
+UPDATE animals
+SET owner_id = 
+    (SELECT id FROM owners WHERE full_name = 'Sam Smith')
+    WHERE name IN ('Agumon');
+
+UPDATE animals
+SET owner_id = 
+    (SELECT id FROM owners WHERE full_name = 'Jennifer Orwell')
+    WHERE name IN ('Gabumon', 'Pikachu');
+
```
Comment (V-Blaze) [inline queries.sql]: > Melody Pond owns Charmander, Squirtle, and Blossom. 
- [ ] You didn't write the code to insert these data above into the DB, and it's definitely going to affect the result of your queries. I'll suggest you add the data and re-run your queries. :+1: :100:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @Dorcas126 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


## Highlights
- Queries are written correctly :heavy_check_mark: 
- Readme looks professional :heavy_check_mark: 
- Well-documented code :heavy_check_mark: 


 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @Dorcas126 , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**